code/main/edge/edgeRequestWorker.brs

```diff
@@ -21,14 +21,29 @@ function _adb_EdgeRequestWorker() as object
         _queue: [],
         _queue_size_max: 50,
 
-        queue: function(requestId as string, xdmData as object, timestampInMillis as longinteger) as void
+        ' ------------------------------------------------------------------------------------------------
+        ' Queue the Edge request.
+        '
+        ' @param requestId          - the request id
+        ' @param xdmEvents          - an array of the XDM events
+        ' @param timestampInMillis  - the timestamp in millis used to compute when to retry failed requets
+        ' @param meta               - the meta data
+        ' @param path               - if it's not empty, overwrite the Edge path with the given value
+        '
+        ' @return void
+        ' ------------------------------------------------------------------------------------------------
+        queue: function(requestId as string, xdmEvents as object, timestampInMillis as longinteger, meta as object, path as string) as void
```

Comment:
Can path be a field inside event instead of a separate field? Same applies for meta.

Reply:
should we create and Event object which contains xdmData, freeFormData (future), ts, request id, path etc. 

```
event : {
  xdm:
  data:
  meta:
  path:
  ...
}
```

Reply:
Either way works for me. It's a good practice to combine multiple parameters into one when the parameter list is too long. In this case, I can't say which option is better. @praveek what do you think?

Reply:
Btw, the `xdmEvents` parameter is changed to be an array of the Edge events, the non-xdm data should be included in the Edge event. So, the combined parameter should be like: 
```json
experienceEvent : {
  xdmEvents: [],
  meta: {},
  path: ""
}
```